src/ResourceManager/Network/Commands.Network/Cortex/HubVnetConnection/RemoveAzureRmHubVirtualNetworkConnectionCommand.cs

```diff
@@ -0,0 +1,136 @@
+            HelpMessage = "The resource id of the hubvirtualnetworkconnection resource to modify.")]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Do not ask for confirmation if you want to overrite a resource")]
+        public SwitchParameter Force { get; set; }
+
+        public override void Execute()
+        {
+            //// Resolve the paramters
+            if (ParameterSetName.Equals(CortexParameterSetNames.ByHubVirtualNetworkConnectionObject, StringComparison.OrdinalIgnoreCase))
+            {
+                var parsedResourceId = new ResourceIdentifier(this.InputObject.Id);
+                this.ResourceGroupName = parsedResourceId.ResourceGroupName;
+                this.ParentResourceName = parsedResourceId.ParentResource.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries).Last();
+                this.Name = parsedResourceId.ResourceName;
+            }
+            else if (ParameterSetName.Equals(CortexParameterSetNames.ByHubVirtualNetworkConnectionResourceId, StringComparison.OrdinalIgnoreCase))
+            {
+                var parsedResourceId = new ResourceIdentifier(this.ResourceId);
+                this.ResourceGroupName = parsedResourceId.ResourceGroupName;
+                this.ParentResourceName = parsedResourceId.ParentResource.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries).Last();
+                this.Name = parsedResourceId.ResourceName;
+            }
+
+            //// Get the virtual hub - this will throw not found if the resource is invalid
+            PSVirtualHub parentVirtualHub = this.GetVirtualHub(this.ResourceGroupName, this.ParentResourceName);
+
+            if (parentVirtualHub == null)
+            {
+                throw new PSArgumentException("The parent hub object could not be found");
+            }
+
+            var connectionToRemove = parentVirtualHub.VirtualNetworkConnections.FirstOrDefault(connection => connection.Name.Equals(this.Name, StringComparison.OrdinalIgnoreCase));
+            if (connectionToRemove != null)
+            {
+                base.Execute();
+
+                ConfirmAction(
+                    Force.IsPresent,
+                    string.Format(Properties.Resources.RemovingResource, Name),
+                    Properties.Resources.RemoveResourceMessage,
+                    this.Name,
+                    () =>
+                    {
+                        parentVirtualHub.VirtualNetworkConnections.Remove(connectionToRemove);
+                        this.CreateOrUpdateVirtualHub(this.ResourceGroupName, this.ParentResourceName, parentVirtualHub, parentVirtualHub.Tag);
+                        WriteObject(true);
```

Comment:
For all removes, you should have a "PassThru" switch parameter, and only output this WriteObject if PassThru is specified: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Aks/Commands.Aks/Commands/RemoveAzureRmAks.cs#L117

Reply:
Fixed